components/PocPrompt.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Text, View, TextInput, TouchableWithoutFeedback, Button, StyleSheet
+} from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    alignItems: 'center'
+  }
+});
+
+// TODO: Impliment api request
+const apiGetRequest = async () => 1500;
+
+// TODO: Impliment api request
+// eslint-disable-next-line no-unused-vars
+const apiPutRequest = async (calorieGoal: number) => { };
+
+export default function PocPrompt() {
+  // State
+  const [displayPrompt, setDisplayPrompt] = useState(false);
+  const [calorieGoal, setCalorieGoal] = useState<number>(undefined);
+
+  // Local Functions
+  const toggleDisplayPrompt = () => {
+    setDisplayPrompt(!displayPrompt);
+  };
+
+  const getGoal = async () => {
+    const goal = await apiGetRequest();
+    setCalorieGoal(goal);
+  };
+
+  const changeGoal = async () => {
+    await apiPutRequest(calorieGoal);
+    setDisplayPrompt(false);
+  };
+
+  // Fetch the initial calorie goal on load
+  useEffect(() => {
```

Comment:
You can probably avoid the lambda wrapper since this is already a parameterless function.

`useEffect(getGoal, []);`

Reply:
It's a weird quirk, but getGoal is async and returns a promise, useEffect only accepts functions that return void, so we have to hide our async invocation in a lambda